src/games/tic_tac_toe.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+exports.setUp = function setUp(players) {
+  let player_sides = {
+    [players[0]]: 'X',
+    [players[1]]: 'O'
+  };
+  let board = [];
+  for (let x = 0; x < 3; y++) {
+    board[x] = [];
+    for (let y = 0; y < 3; x ++) {
+      board[x][y] = {
+        position: {x, y}
+      };
+    }
+  }
+  return {
+    players,
+    player_sides,
+    board,
+    turn: players[0]
+  }
+}
+
+function checkWinner(board) {
+  for (let i = 0; i < 3; i++) {
+    row_owner = [];
+    col_owner = [];
+    for (let j = 0; j < 3; j++) {
+      row_owner[j] = board[i][j].owner;
+      col_owner[j] = board[j][i].owner;
+    }
+    for (const arr of [row_owner, col_owner]) {
+      if (arr[0] !== null && arr[0] != undefined) {
+        if (arr.every( (val, i, arr) => val === arr[0])) {
+          return arr[0];
+        }
+      }
+    }
+  }
+  diag1_owner = [board[0][0].owner, board[1][1].owner, board[2][2].owner];
+  diag2_owner = [board[0][2].owner, board[1][1].owner, board[2][0].owner];
+  for (const arr of [diag1_owner, diag2_owner]) {
+    if (arr[0] !== null && arr[0] != undefined) {
+      if (arr.every( (val, i, arr) => val === arr[0])) {
+        return arr[0];
+      }
+    }
+  }
+  return null;
+}
+
+function otherPlayer(state, player) {
+  if (player === state.players[0]) {
+    return state.players[1];
+  } else {
+    return state.players[0];
+  }
+}
+
+exports.transition = function transition(state, action) {
+  if (action.player != state.turn) {
+    return state;
+  }
+  let {x, y} = action.args.position;
+  state.board[x][y] = {
```

Comment:
Which makes me think we don't necessarily have a very nice way to describe that the move is invalid back into the scene as of now.